### PR TITLE
refactor: replace some filter()[0] usages to avoid unneeded iterations

### PR DESCRIPTION
**Pull request by makepanic**
This probably won't affect performance in the common case.
This removes just some unnecessary array method calls.

### CODE REVIEW

File: lib/orm/model.js [hunk 1]
```diff
@@ -824,15 +824,18 @@ class Model {
...
+            // filter out models that will still be associated
+            !tempAssociation.includes(associate) &&
+            associate.hasInverseFor(association)
+        )
         .forEach(associate => {
-          if (associate.hasInverseFor(association)) {
             let inverse = associate.inverseFor(association);
 
             associate.disassociate(this, inverse);
             associate.save();
```
Comment (jelhan) [inline lib/orm/model.js]: Indentation must be adjusted after `if` block is removed.

File: lib/orm/polymorphic-collection.js [hunk 2]
```diff
@@ -118,7 +118,7 @@ export default class PolymorphicCollection {
    * @public
    */
   includes(model) {
-    return this.models.filter(m => isEqual(m.attrs, model.attrs)).length > 0;
+    return this.models.find(m => isEqual(m.attrs, model.attrs)) !== undefined;
```
[the remark this answers is not in the document]
Comment (makepanic) [reply lib/orm/polymorphic-collection.js]: Changed it to some as it makes it easier to understand.

File: lib/orm/model.js (hunk 1 above)
Comment (makepanic) [reply lib/orm/model.js]: You're right. I didn't see that prettier is usable in this project. I ran prettier on the changed files and rebase + pushed the changes.

Comment (samselikoff) [reply lib/orm/model.js]: thanks for the review @jelhan 😄 Very helpful to have other folks checking this out!

@makepanic I think I did install prettier here but still need to (a) go through and prettierify everything and (b) figure out the best way to enforce this (husky? tests?)

Comment (makepanic) [reply lib/orm/model.js]: We're using eslint with https://github.com/prettier/eslint-plugin-prettier which fails the linting stage if prettier has issues with our code.

Comment (makepanic) [reply lib/orm/model.js]: For running prettier on all project files, you could add a package.json script like:
```
    "prettier": "prettier --write \"{config,app,tests}/**/*.{js,ts}\"",
```
And adjust it to the mirage source files.

**Comment by makepanic** [comment]
rebased this onto current master

**Comment by ryanto** [comment]
Thanks @makepanic !

**Pull request merged**